Parser/Startup.cs

```diff
@@ -23,16 +16,22 @@ public Startup(IConfiguration configuration)
         }
 
         public IConfiguration Configuration { get; }
-
-        // This method gets called by the runtime. Use this method to add services to the container.
+        
         public void ConfigureServices(IServiceCollection services)
         {
-            services.AddScoped<IDataProvider, Requester>();
-            services.AddScoped<IParser, HTMLParser>();
+            //  DefaultHtmlProvider must be the last 
```

Comment:
If you rely on the order of injected dependencies it means your design could be improved.